libcore/test/catch2/geometry/GeometryHelperTest.cpp

```diff
@@ -26,580 +26,998 @@
+                REQUIRE(wallItr->GetPoint2() == std::next(wallItr)->GetPoint1());
             }
 
-            SECTION("Door on not neighboring wall elements")
-            {
-                for(size_t i = 0; i < trackWalls.size(); ++i) {
-                    for(size_t j = 0; j < trackWalls.size(); ++j) {
-                        if(std::abs(static_cast<int>(i) - static_cast<int>(j)) <= 1) {
-                            continue;
-                        }
-                        Transition trainDoor;
-                        trainDoor.SetPoint1(trackWalls[i].GetCentre());
-                        trainDoor.SetPoint2(trackWalls[j].GetCentre());
-
-                        std::pair<std::pair<Point, Wall>, std::pair<Point, Wall>>
-                            wallDoorIntersection = {
-                                {trainDoor.GetPoint1(), trackWalls[i]},
-                                {trainDoor.GetPoint2(), trackWalls[j]}};
-
-                        auto [addedWalls, removedWalls] = geometry::helper::SplitWall(
-                            wallDoorIntersection, trackWalls, trainDoor);
-
-                        Wall newWall1, newWall2;
-
-                        if(i < j) {
-                            newWall1 = {trackWalls[i].GetPoint1(), trainDoor.GetPoint1()};
-                            newWall2 = {trackWalls[j].GetPoint2(), trainDoor.GetPoint2()};
-                        } else {
-                            newWall1 = {trackWalls[j].GetPoint1(), trainDoor.GetPoint2()};
-                            newWall2 = {trackWalls[i].GetPoint2(), trainDoor.GetPoint1()};
-                        }
-
-                        REQUIRE(addedWalls.size() == 2);
-                        REQUIRE_THAT(addedWalls, Catch::VectorContains(newWall1));
-                        REQUIRE_THAT(addedWalls, Catch::VectorContains(newWall2));
-
-                        unsigned int numRemovedWalls =
-                            1 + std::abs(static_cast<int>(i) - static_cast<int>(j));
-                        REQUIRE(removedWalls.size() == numRemovedWalls);
-
-                        for(auto k = std::min(i, j); k <= std::max(i, j); ++k) {
-                            REQUIRE_THAT(removedWalls, Catch::VectorContains(trackWalls[k]));
-                        }
-                    }
-                }
+            std::vector<Wall> trackWallsReversed{
+                trackWall5, trackWall4, trackWall3, trackWall2, trackWall1};
+
+            REQUIRE_NOTHROW(geometry::helper::SortWalls(trackWallsReversed, trackStart));
+            CHECK_THAT(trackWallsReversed, Catch::Equals(trackWallsOrdered));
+            for(auto wallItr = std::begin(trackWallsReversed);
+                wallItr != std::end(trackWallsReversed) - 1;
+                ++wallItr) {
+                REQUIRE(wallItr->GetPoint2() == std::next(wallItr)->GetPoint1());
             }
         }
 
-        SECTION("unordered track walls")
+        SECTION("non-continuous wall")
```

Comment:
what's the difference to the test starting at line 436?

Reply:
no difference in functionality, removed redundant test